glotaran/analysis/residual_calculation.py

```diff
@@ -82,14 +80,12 @@ def apply_weight(matrix, weight):
             residuals[label].append(residual)
             penalties.append(residual)
 
-            if (
-                callable(scheme.model.has_additional_penalty_function)
-                and scheme.model.has_additional_penalty_function()
-            ):
-                additional_penalty = dask.delayed(scheme.model.additional_penalty_function)(
-                    parameter, clp_label, clp, i
-                )
-                penalties.append(additional_penalty)
+            if callable(scheme.model.has_additional_penalty_function):
```

Comment:
This call needs to not happen inside the for loop, but at  the same level, after the clps are calculated. It then uses the clps and the penalty model spec to calculate the area difference (area1 - penalty.parameter*area2)*penatly.weight and adds that (single number per penalty) to the residual vector.